### PR TITLE
hooks lint part 1

### PR DESCRIPTION
**Pull request by chrisnojima**
This PR fixes all lint disabled lines that explicitly turn off the exhaustive rule. I'll do the one that aren't so explicit next

### CODE REVIEW

File: shared/wallets/send-form/participants/to-field.tsx [hunk 1]
```diff
@@ -82,30 +80,11 @@ export type ToStellarPublicKeyProps = {
 }
 
 const ToStellarPublicKey = (props: ToStellarPublicKeyProps) => {
-  const [recipientPublicKey, setRecipentPublicKey] = React.useState(props.recipientPublicKey)
```
Comment (jakob223) [inline shared/wallets/send-form/participants/to-field.tsx]: I don't really understand what happened here. What's the deal with the old debounced code / why is it no longer needed?

Comment (chrisnojima) [reply shared/wallets/send-form/participants/to-field.tsx]: this convoluted code was to try and rectify / debounce scanning qr codes on android. syncing the props / state doing this is really flakey and we should just fix it in another way

File: shared/chat/conversation/bot/team-picker.tsx [hunk 2]
```diff
@@ -35,7 +34,8 @@ const BotTeamPicker = (props: Props) => {
         logger.info('BotTeamPicker: error loading search results: ' + error.message)
       }
     )
-  }
+  }, [term, submit])
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/chat/conversation/bot/team-picker.tsx]: those are always the same so they can be omitted. the lint is aware of this

File: shared/wallets/send-form/participants/to-field.tsx (hunk 1 above)
Comment (jakob223) [reply shared/wallets/send-form/participants/to-field.tsx]: Do we have a ticket for that?

Comment (chrisnojima) [reply shared/wallets/send-form/participants/to-field.tsx]: HOTPOT-1810

**Pull request merged**